src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -382,11 +421,11 @@ template<
 void GAN<Model, InitializationRuleType, Noise, PolicyType>::Forward(
     arma::mat&& input)
 {
-  if (!reset)
+  if (parameter.is_empty())
     Reset();
 
   generator.Forward(std::move(input));
-  ganOutput = boost::apply_visitor(
+  arma::mat ganOutput = boost::apply_visitor(
```

Comment:
Looks like this could be const.

Reply:
It looks like the `FFN'::Forward()` argument is not `const`. It is throwing an error. So, I have kept it as it is.